src/directives/rcdni.js

```diff
@@ -0,0 +1,42 @@
+directives.directive('rcDni',['dniSemanticsValidationService','dniSyntaxValidationService',function(dniSemanticsValidationService,dniSyntaxValidationService){
+	return{
+		require: 'ngModel',					
+		link: function (scope, elm, attrs, ctrl){	
+			var validationToExecute = function (viewValue){
+				var currentTypedValue = viewValue;
+				validateDNI(currentTypedValue);
+			};
+
+			function validateDNI(dni){
+				if (validateDNISyntax(dni)){
```

Comment:
That was originally the case, but I changed it because Evaristo precisely identified that if one of the two validations is wrong, then it does not really matter what the other does, and that makes sense. That said, I believe it is perfectly valid to change it as per Jesus' suggestion. Should I refactor it again then? I'm fine either way.